src/charm.py

```diff
@@ -117,6 +123,55 @@ def _on_config_changed(self, event: ops.HookEvent) -> None:
         """
         self.change_config(event)
         self._set_workload_version()
+        if self._charm_state.enable_mjolnir:
+            self._enable_mjolnir()
+
+    def _enable_mjolnir(self) -> None:
+        """Enable mjolnir service."""
+        container = self.unit.get_container(SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            self.unit.status = ops.MaintenanceStatus("Waiting for pebble")
+            return
+        self.model.unit.status = ops.MaintenanceStatus("Configuring Mjolnir")
+        try:
+            admin_username = token_hex(16)
+            admin_user = actions.register_user(container, admin_username, True)
+            admin_access_token = admin_user.access_token
+            mjolnir_user = actions.register_user(
+                container,
+                MJOLNIR_USER,
+                True,
+                str(self._charm_state.server_name),
+                admin_access_token,
+            )
+            # Create (or get) the management room
+            # Add the bot to the management room if we are creating it
+            mjolnir_access_token = mjolnir_user.access_token
+            room_id = synapse.get_room_id(room_name="management", access_token=admin_access_token)
+            synapse.create_mjolnir_config(
+                container=container, access_token=mjolnir_access_token, room_id=room_id
+            )
+            synapse.override_rate_limit(
+                user=mjolnir_user, access_token=admin_access_token, charm_state=self._charm_state
+            )
+            synapse.deactivate_user(
+                user=admin_user,
+                server=str(self._charm_state.server_name),
+                access_token=admin_access_token,
+            )
+            self.pebble_service.replan_mjolnir(container)
+        except synapse.WorkloadError as exc:
+            logger.exception("Failed to interact with Synapse workload: %r", exc)
+            self.model.unit.status = ops.BlockedStatus(str(exc))
+            return
+        except synapse.APIError as exc:
+            logger.exception("Failed to interact with Synapse API: %r", exc)
+            self.model.unit.status = ops.BlockedStatus(str(exc))
+            return
+        except actions.RegisterUserError as exc:
+            logger.exception("Failed to register Mjolnir user: %r", exc)
+            self.model.unit.status = ops.BlockedStatus(str(exc))
```

Comment:
When a charm is in blocked state we expect an action from the user to solve the problem.
If we don't expect any action from the user, and it's a temporary error, don't put in blocked state and either retry or expect another event to retry that action
If we don't expect any action from the user, and the error is unrecoverable, it's preferable to let the exception go through
If we expect an action from the user, it's probably better to state what they should do in the BlockedStatus message

Reply:
Nice. Now I changed to block only if the management room does not exist ( I'm considering for now that will be created by an admin user)